### PR TITLE
Feature/generic channel aspect

### PR DESCRIPTION
**Pull request by szuecs**
GenericChannelAspect can be used to send arbitrary string float64 pairs to this monitoring tool.
It will aggregate float64 values grouped by the given string.

A general change is that we expose Count in all Aspects in which it makes sense.

### CODE REVIEW

**Comment by szuecs** [comment]
retrigger

File: aspects/counter_test.go [hunk 1]
```diff
@@ -109,24 +108,24 @@ func Test_InRoot(t *testing.T) {
...
+// 	}
+// 	ca.increment(tup)
+// 	ca.reset()
+
+// 	router.Use(CounterHandler(ca))
+// 	tryRequest(router, "GET", "/")
+// 	if assert.Equal(t, expect, ca.RequestsSum, "Incrementation of counter does not work, expect %d but got %d %s", expect, ca.RequestsSum, ballotX) {
+// 		t.Logf("CounterHandler works, expect %d and got %d %s", expect, ca.RequestsSum, checkMark)
+// 	}
+// }
```
Comment (mikkeloscar) [inline aspects/counter_test.go]: I guess this should just be deleted if it's not used.

File: aspects/generic_channel.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+			gc.calculate()
+		}
+	}()
+}
+
+// SetupGenericChannelAspect returns an unbuffered channel for type
+// DataChannel, such that you can send arbitrary key (string) value
+// (float64) pairs to it.
+func (gc *GenericChannelAspect) SetupGenericChannelAspect() chan DataChannel {
+	_gc := gc // save gc in closure
```
Comment (Raffo) [inline aspects/generic_channel.go]: I would rename this variable. "_" in go has a very well defined meaning and I would not mess with it.

**Comment by Raffo** [comment]
I have a quick look, but apart from the two top comments, looks good.

**Comment by szuecs** [comment]
@Raffo @mikkeloscar done

**Comment by Raffo** [comment]
👍

**Comment by szuecs** [comment]
:+1:

**Pull request merged**